src/fd_cache.nim

```diff
@@ -0,0 +1,310 @@
+##
+## Copyright (c) 2024, Crash Override, Inc.
+##
+## This file is part of Chalk
+## (see https://crashoverride.com/docs/chalk)
+##
+
+##
+## FD cache
+## Global cache for opened files
+##
+## The overall concept is that there is a global cache
+## for all opened file streams.
+##
+## Some things you can do:
+## * yield   - create or get existing file stream from cache.
```

Comment:
'yield' generally means "I'm done with this (for now)". Generally, I would take that to mean the system can decide whether it needs to be closed (whereas close generally asserts we need to close).   'Acquire' is the right word to use here.

Reply:
was going between the 2 but yeah the logic for acquire makes more sense

Reply:
renamed in https://github.com/crashappsec/chalk/pull/229/commits/411b868deda805fca743cd1511845ad9adc44662